opm/grid/cpgrid/Entity.hpp

```diff
@@ -529,6 +524,11 @@ Dune::cpgrid::Geometry<3,3> Dune::cpgrid::Entity<codim>::geometryInFather() cons
     if (!(this->hasFather())){
         OPM_THROW(std::logic_error, "Entity has no father.");
     }
+    // static    to not need any extra storage per Enitity. One object used for all instances
+    // constexpr to allow for in-class instantiation
+    /// \brief Indices of corners in entity's geometry in father reference element, for the Geometry returned by geometryInFather
+    static constexpr std::array<int,8> in_father_reference_elem_corner_indices_ = {0,1,2,3,4,5,6,7};
```

Comment:
Nice.

The only thing that I am wondering about is whether we really need the static. Should we maybe drop it?

This is a local variable now. Hence the storage argument is a bit different. For a local variable, that is not static, the storage is freed after the function call. So storage is only used during this function. By making it static, the variable (and storage) is allocated and initialized at program start and stays until the end.
One reason to make it static would be if the initialization is costly. This does not hold here.

Reply:
Thanks for the explanation. Dropping it creates a warning (storing the address of a local variable in the return value of geometryInFather()), so I would keep it. 

I fixed the documentation too

Reply:
I see. Did not look how it is used. Well the function name should have told me. Please add documentation that the Geometry that we return stores a pointer to this and that therefore we need to make sure that the lifetime goes beyond this function.

Reply:
Comment has been added, thanks!